Ginger/GingerCore/ALM/OctaneCore.cs

```diff
@@ -226,5 +365,520 @@ public override bool IsServerConnected()
+                                if (stepActivityVarOptionalVar == null)
+                                {
+                                    //no such variable value option so add it
+                                    stepActivityVarOptionalVar = new OptionalValue(paramSelectedValue);
+                                    ((VariableSelectionList)stepActivityVar).OptionalValuesList.Add(stepActivityVarOptionalVar);
+                                    if (isflowControlParam == true)
+                                        stepActivity.AutomationStatus = eActivityAutomationStatus.Development;//reset status because new param value was added
+                                }
+                                //set the selected value
+                                ((VariableSelectionList)stepActivityVar).SelectedValue = stepActivityVarOptionalVar.Value;
+                            }
+                            else
+                            {
+                                //try just to set the value
+                                try
+                                {
+                                    stepActivityVar.Value = paramSelectedValue;
+                                    if (stepActivityVar is VariableString)
+                                        ((VariableString)stepActivityVar).InitialStringValue = paramSelectedValue;
+                                }
+                                catch (Exception ex) { Reporter.ToLog(eLogLevel.ERROR, $"Method - {MethodBase.GetCurrentMethod().Name}, Error - {ex.Message}", ex); }
+                            }
+
+                            //add linked variable if needed
+                            if (string.IsNullOrEmpty(linkedVariable) == false)
+                            {
+                                stepActivityVar.LinkedVariableName = linkedVariable;
+                            }
+                            else
+                                stepActivityVar.LinkedVariableName = string.Empty;//clear old links
+                        }
+                    }
+
+                    //order the Activities Group activities according to the order of the matching steps in the TC
+                    try
+                    {
+                        int startGroupActsIndxInBf = busFlow.Activities.IndexOf(tcActivsGroup.ActivitiesIdentifiers[0].IdentifiedActivity);
+                        foreach (QC.QCTSTestStep step in tc.Steps)
+                        {
+                            int stepIndx = tc.Steps.IndexOf(step) + 1;
+                            ActivityIdentifiers actIdent = (ActivityIdentifiers)tcActivsGroup.ActivitiesIdentifiers.Where(x => x.ActivityExternalID == step.StepID).FirstOrDefault();
+                            if (actIdent == null || actIdent.IdentifiedActivity == null) break;//something wrong- shouldn't be null
```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6317725)